### ISSUE TITLE
Cosigning servers policy and wrong spend transaction

### ISSUE THREAD

**Issue by darosior** [re-vault/practical-revault#4]
TL;DR: we currently assume authorized spenders (or more precisely, their wallet) are not going to submit a wrong tx to the cosigning servers, are we ok with that?

The cosigning server, as currently specified in `messages.md` will refuse to sign an input spending a specific unvault transaction twice.

However, if the watchtowers refuse the spending attempt the "authorized spenders" are stuck with the already-signed spend transaction for the entire life of this vault.

This is not *that* bad, since the spend transaction is (hopefully!) valid and can be used anyway. However this restrains usability in case of mistake on their end.

**Comment by darosior** [comment]
I think that's an UX problematic: each of them is signing it, we should display them the fees (maybe compare them to the current market?) and they ACK it by signing it.

Pinging @edouardparis for the UI / UX above part.

@kloaec we could otherwise make the spend transaction signal for RBF and let the cosigning server re-sign the *same* transaction with just more fees. But i don't really like this idea.

**Comment by kloaec** [comment]
Not sure why you talk about fees here. Please clarify :) (IMHO Feebumping is external to this)
Regarding the invalid-policy transaction being signed, I think it's a feature. If they want to actually push it, revault (should) happen.
UX-wise this revaulting should be notified to the Stakeholders to figure out wtf happened.

**Comment by darosior** [comment]
> Not sure why you talk about fees here. Please clarify :) (IMHO Feebumping is external to this)

Not related to our fee-bumping strategy for revocation transactions.
I'm talking about the fees of the spend transaction for actual operation practicality: remember that the spend transaction is presigned N blocks before actually being able to broadcast it. The feerate might have mooned during this locktime period and the fund managers are doomed to wait 24 hours before actually seeing their transaction confirmed, while they'd happily have spent a few more sats for bumping the fees.
I think that's a UX regression for our users.


> Regarding the invalid-policy transaction being signed, I think it's a feature. If they want to actually push it, revault (should) happen.

Yeah definitely. The mention of policy in the title refers to the sign-only-once policy. The point is about the practicality of having pre-signed non-bumpable valid spend tx for the fund managers. Even worse if they change their mind (unlikely as they all sign, but hey).
